dashboard/app/models/user_proficiency.rb

```diff
@@ -0,0 +1,66 @@
+class UserProficiency < ActiveRecord::Base
+  belongs_to :user
+
+  # The set of concepts for which proficiency is computed.
+  # NOTE: This list is tied to DB columns. DO NOT EDIT without a DB migration.
+  CONCEPTS = %w(
+    sequencing
+    debugging
+    repeat_loops
+    repeat_until_while
+    for_loops
+    events
+    variables
+    functions
+    functions_with_params
+    conditionals
+  )
+  # The maximum difficulty ranking for a concept.
+  # NOTE: This number is tied to DB columns. DO NOT EDIT without a DB migration.
+  MAXIMUM_CONCEPT_DIFFICULTY = 5
+
+  # Returns the number of levels the user has shown proficiency in for the
+  # indiciated concept and difficulty or higher.
+  def get_level_count(concept, difficulty_number)
+    return 0 if !CONCEPTS.include? concept
+    return 0 if !(1..MAXIMUM_CONCEPT_DIFFICULTY).cover? difficulty_number
+
+    num_levels = 0
+    (difficulty_number..MAXIMUM_CONCEPT_DIFFICULTY).each do |d|
+      field_name = concept + "_" + "d" + d.to_s + "_" + "count"
```

Comment:
And conveniently you don't need a to_s because string interpolation does
that for you
On Apr 7, 2016 3:47 PM, "Elijah Hamovitz" <notifications@github.com> wrote:

> In dashboard/app/models/user_proficiency.rb
> <https://github.com/code-dot-org/code-dot-org/pull/7748#discussion_r58957967>
> :
>
> > +    functions_with_params
> > +    conditionals
> > +  )
> > +  # The maximum difficulty ranking for a concept.
> > +  # NOTE: This number is tied to DB columns. DO NOT EDIT without a DB migration.
> > +  MAXIMUM_CONCEPT_DIFFICULTY = 5
> > +
> > +  # Returns the number of levels the user has shown proficiency in for the
> > +  # indiciated concept and difficulty or higher.
> > +  def get_level_count(concept, difficulty_number)
> > +    return 0 if !CONCEPTS.include? concept
> > +    return 0 if !(1..MAXIMUM_CONCEPT_DIFFICULTY).cover? difficulty_number
> > +
> > +    num_levels = 0
> > +    (difficulty_number..MAXIMUM_CONCEPT_DIFFICULTY).each do |d|
> > +      field_name = concept + "_" + "d" + d.to_s + "_" + "count"
>
> I think "#{concept}_d#{d.to_s}_count" is preferred
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/7748/files/247d1b0931dca6314b94f51ed5c0f2118ba17158#r58957967>
>